Jenkinsfile

```diff
@@ -1,111 +1,131 @@
+                    """
+                }
+            }
+        }
+        stage("Checkout config_ocs") {
             steps {
-                // Direct the HOME to WORKSPACE for pip to get the
-                // installed library.
-                // 'PATH' can only be updated in a single shell block.
-                // We can not update PATH in 'environment' block.
-                // Pytest needs to export the junit report.
-                withEnv(["HOME=${env.WORKSPACE}"]) {
+                script {
                     sh """
-                        source /home/saluser/.setup_dev.sh || echo loading env failed. Continuing...
-                        pip install .[dev]
-                        setup -k -r .
-                        pytest --cov-report html --cov=${env.MODULE_NAME} --junitxml=${env.XML_REPORT}
+                    docker exec -u saluser \${container_name} sh -c \"source ~/.setup.sh && cd /home/saluser/repos/ts_config_ocs/ && /home/saluser/.checkout_repo.sh \${work_branches} \"
                     """
                 }
             }
         }
-        stage('Build and Upload Documentation') {
+        stage("Running tests") {
             steps {
-                withEnv(["HOME=${env.WORKSPACE}"]) {
+                script {
                     sh """
-                    source /home/saluser/.setup_dev.sh || echo loading env failed. Continuing...
-                    pip install .[dev]
-                    package-docs build
-                    ltd upload --product ts-hvac --git-ref ${GIT_BRANCH} --dir doc/_build/html
+                    docker exec -u saluser \${container_name} sh -c \"source ~/.setup.sh && cd repo && pip install --ignore-installed -e . && eups declare -r . -t saluser && setup ts_hvac -t saluser && pytest --junitxml=\${XML_REPORT}\"
                     """
                 }
             }
         }
     }
-
     post {
         always {
-            // Change the ownership of workspace to Jenkins for the clean up
-            // This is a "work around" method
-            withEnv(["HOME=${env.WORKSPACE}"]) {
-                sh 'chown -R 1003:1003 ${HOME}/'
-            }
-
-            // The path of xml needed by JUnit is relative to
-            // the workspace.
-            junit 'jenkinsReport/*.xml'
-
             // Publish the HTML report
             publishHTML (target: [
                 allowMissing: false,
                 alwaysLinkToLastBuild: false,
                 keepAll: true,
-                reportDir: 'htmlcov',
+                reportDir: 'jenkinsReport/',
                 reportFiles: 'index.html',
                 reportName: "Coverage Report"
               ])
-        }
 
+            sh "docker exec -u saluser \${container_name} sh -c \"" +
+                "source ~/.setup.sh && " +
+                "cd /home/saluser/repo/ && " +
+                "setup ts_salobj -t saluser && " +
+                "package-docs build\""
```

Comment:
Same question as for ts_MTDome: shouldn't you be setting up ts_hvac and be in that dir to build the docs?

Reply:
Yes, as in ts_MTDome there is a copy and paste error there. However, it is correct to do this in the `repo` directory.

Reply:
I am really surprised. I always run `package-docs build` inside my package directory. I wonder what it does if you run it a level up? I suspect it would build docs for all packages in `repo` instead of just the one I care about.

Reply:
If you have a look at the rest of the Jenkinsfile then `repo` is used everywhere as the name of the directory where the code has been checked out. It is confusing since there indeed also is a `repos` directory.

Reply:
Ah, thanks. So /home/saluser/repo/ is the name of the package, and all other packages are in cd /home/saluser/repos? That certainly is confusing!

Reply:
Exactly. Very confusing!